miasm2/jitter/codegen.py

```diff
@@ -353,24 +357,28 @@ def gen_goto_code(self, attrib, instr_offsets, dst):
         @attrib: instruction Attributes
         @instr_offsets: instructions offsets list
         @dst: potential instruction destination"""
-        if isinstance(dst, AsmLabel) and dst.offset is None:
-            # Generate goto for local labels
-            return ['goto %s;' % dst.name]
-        offset = None
-        if isinstance(dst, AsmLabel) and dst.offset is not None:
-            offset = dst.offset
-        elif isinstance(dst, (int, long)):
-            offset = dst
+
         out = []
-        if (offset is not None and
-            offset > attrib.instr.offset and
+        if isinstance(dst, Expr):
+            out += self.gen_post_code(attrib)
+            out.append('BlockDst->address = DST_value;')
+            out += self.gen_post_instr_checks(attrib)
+            out.append('\t\treturn JIT_RET_NO_EXCEPTION;')
+            return out
+
+        assert isinstance(dst, LocKey)
+        offset = self.ir_arch.symbol_pool.loc_key_to_offset(dst)
+        name = self.ir_arch.symbol_pool.loc_key_to_name(dst)
+        if offset is None:
+            # Generate goto for local labels
+            return ['goto %s;' % name]
```

Comment:
As `name` is used only here, `name = self.ir_arch.symbol_pool.loc_key_to_name(dst)` can be moved inside the `if` statement

Reply:
done